### PR TITLE
OCPBUGS-5064: E2E: Per Core Runtime Tuning Test automation

### PR DESCRIPTION
**Pull request by mrniranjan**
Signed-off-by: Niranjan M.R <mrniranjan@redhat.com>

### CODE REVIEW

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (marioferh) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:828]: This test override this one right? 
https://github.com/marioferh/cluster-node-tuning-operator/blob/5d9c2e11ad5595b7724edb327ed1a34e83518fc1/test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go#L638

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (marioferh) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:911]: And this one overrides this: https://github.com/marioferh/cluster-node-tuning-operator/blob/5d9c2e11ad5595b7724edb327ed1a34e83518fc1/test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go#L676

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (marioferh) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:997]: this test must fail, where we are catching the expected error?

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (marioferh) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1015]: this test is of https://github.com/cri-o/cri-o/pull/5927, @bartwensley PTAL?

**Review by marioferh — COMMENTED** [review]
some comments

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:828]: Yes, this test we are checking if we can move from one Hint to another is the system tuned appropriately

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:911]: Yes, this test we are checking if we can move from one Hint to another is the system tuned appropriately

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:997]: yes this test fails, we are checking if it fails.  May be i should check the error

Comment (marioferh) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:997]: Yep that's what I mean

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1039]: What will the default c-state/p-state configuration be for the server running the test? Just wondering if these will have any impact if the system is configured for low power by default?

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1093]: After the pod is deleted, it would be good to check that the cpus that were used by the pod have been set back to their original pm_qos_resume_latency and scaling_governor.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1078]: Maybe it is too much, but I wonder if it would be useful to check that cpus not in use by the pod did not have their pm_qos_resume_latency_us or scaling_governor modified?

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1096]: The comments from the previous testcase would apply to this testcase as well.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1039]: @bartwensley  In this case system configuration is what the perPodPowerManagement configures. so intel_pstate is set passive. And the bios is set it's power settings to OS controlled.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1093]: ack, will add that

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1078]: ack will check that

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1039]: OK - and I assume c-states are also enabled in the BIOS?

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1106]: @bartwensley i have addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1077]: @bartwensley i have addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1098]: @bartwensley i have addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1038]: @bartwensley Yes C-states are enabled in BIOS

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:998]: @marioferh Addressed this on latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1014]: nit: spelling - parameters and annotations

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1851]: nit: a better name might be checkCpuGovernorsAndResumeLatency

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1177]: This should be down one line. It should also say "have default settings".

**Review by bartwensley — CHANGES_REQUESTED** [review]

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @bartwensley  addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @bartwensley addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @bartwensley addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1176]: @bartwensley addressed this in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (marioferh) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:829]: Then we can remove the previous one?

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (marioferh) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:998]: thx

**Review by marioferh — COMMENTED** [review]
/lgtm

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1176]: This is incorrect - the code is not checking whether the remaining cpus "donot have powersave settings" - it is checking that they have the default settings, which are a mix (c-states are enabled and governor is performance).

**Review by bartwensley — CHANGES_REQUESTED** [review]

**Review by bartwensley — APPROVED** [review]
Looks good.

**Comment by bartwensley** [comment]
> @bartwensley @marioferh request to review , I have made minor changes where tests related to pod annotations are skipped if systems are not numa aware. These test cases need to be run on Real hardware where bios tuning is done with regard to power management.

When you are making minor changes to the review, it would be much easier for the reviewers if you did these in a new commit (and then squash before merging).

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1034]: Can you move this to a utility function so it doesn't need to be repeated below? I can imagine other test cases may want to use this in the future as well.

Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1034]: Addressed this in latest commit

**Comment by mrniranjan** [comment]
> > @bartwensley @marioferh request to review , I have made minor changes where tests related to pod annotations are skipped if systems are not numa aware. These test cases need to be run on Real hardware where bios tuning is done with regard to power management.
> 
> When you are making minor changes to the review, it would be much easier for the reviewers if you did these in a new commit (and then squash before merging).

agree apologize for this.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1893]: Do the test systems all really have 80 or more cpus? That seems high.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (bartwensley) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1894]: The string doesn't match the check. Could use a variable or constant so the string always matches the check.

Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: addressed in the latest commit.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: The systems that we work  like Dell R740 has 80cpus

Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: okay, i will reduce it to 32 cpus

Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: addressed this in the latest commit

**Comment by yanirq** [comment]
@mrniranjan   please rebase this PR on top of latest master.  Some of the tests have been modified and we want to get the right state of e2e-gcp-pao

**Comment by yanirq** [comment]
@mrniranjan  please rebase this first to make sure its under v2 updates

**Comment by mrniranjan** [comment]
@yanirq can you check the latest ci results and also the changes i have added a new commit to the changes done . So if its good i will squash the commits.

**Comment by yanirq** [comment]
@mrniranjan  please rebase this PR first on top of the latest changes ,  I would like to see an updated test run

**Comment by mrniranjan** [comment]
@yanirq can we remove the Do not merge/hold flag ?

**Pull request merged**
